src/main/java/org/logstash/dissect/fields/AbstractField.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.dissect.fields;
+
+import com.logstash.Event;
+import org.logstash.dissect.Delimiter;
+import org.logstash.dissect.ValueResolver;
+
+import java.util.Map;
+
+public abstract class AbstractField implements Field {
+    private final int ordinal;
+    protected final String name;
+
+    protected Delimiter join;
+    protected Delimiter next;
```

Comment:
I found the problem. the protected was used to allow subclass `toString` method to build a more informative string.
So I moved the StringBuilder to the super class and called it with the name of the calling class